src/traces/parcats/attributes.js

```diff
@@ -0,0 +1,202 @@
+var domainAttrs = require('../../plots/domain').attributes;
+var scatterAttrs = require('../scatter/attributes');
+var scatterLineAttrs = scatterAttrs.line;
+var colorbarAttrs = require('../../components/colorbar/attributes');
+
+var line = extendFlat({
+    editType: 'calc'
+}, colorAttributes('line', {editType: 'calc'}),
+    {
+        showscale: scatterLineAttrs.showscale,
+        colorbar: colorbarAttrs,
+        shape: {
+            valType: 'enumerated',
+            values: ['linear', 'hspline'],
+            dflt: 'linear',
+            role: 'info',
+            editType: 'plot',
+            description: [
+                'Sets the shape of the paths.',
+                'If `linear`, paths are composed of straight lines.',
+                'If `hspline`, paths are composed of horizontal curved splines'
+            ].join(' ')
+        }
+    });
+
+module.exports = {
+    domain: domainAttrs({name: 'parcats', trace: true, editType: 'calc'}),
+    hoverinfo: extendFlat({}, plotAttrs.hoverinfo, {
+        flags: ['count', 'probability'],
+        editType: 'plot'
+        // plotAttrs.hoverinfo description is appropriate
+    }),
+    hoveron: {
+        valType: 'enumerated',
+        values: ['category', 'color', 'dimension'],
+        dflt: 'category',
+        role: 'info',
+        editType: 'plot',
+        description: [
+            'Sets the hover interaction mode for the parcats diagram.',
+            'If `category`, hover interaction take place per category.',
+            'If `color`, hover interactions take place per color per category.',
+            'If `dimension`, hover interactions take place across all categories per dimension.'
+        ].join(' ')
+    },
+    arrangement: {
+        valType: 'enumerated',
+        values: ['perpendicular', 'freeform', 'fixed'],
+        dflt: 'perpendicular',
+        role: 'style',
+        editType: 'plot',
+        description: [
+            'Sets the drag interaction mode for categories and dimensions.',
+            'If `perpendicular`, the categories can only move along a line perpendicular to the paths.',
+            'If `freeform`, the categories can freely move on the plane.',
+            'If `fixed`, the categories and dimensions are stationary.'
+        ].join(' ')
+    },
+    bundlecolors: {
+        valType: 'boolean',
+        dflt: true,
+        role: 'info',
+        editType: 'plot',
+        description: 'Sort paths so that like colors are bundled together within each category.'
+    },
+    sortpaths: {
+        valType: 'enumerated',
+        values: ['forward', 'backward'],
+        dflt: 'forward',
+        role: 'info',
+        editType: 'plot',
+        description: [
+            'Sets the path sorting algorithm.',
+            'If `forward`, sort paths based on dimension categories from left to right.',
+            'If `backward`, sort paths based on dimensions categories from right to left.'
```

Comment:
I still think we can do better with sorting in general. Take for example `parcats_bundled`:
![screen shot 2018-09-28 at 11 22 31 am](https://user-images.githubusercontent.com/2678795/46217748-d8dc2800-c310-11e8-899b-11743a11666f.png)
In the `Three` dimension all the grey paths come together, yet the B->11 and C->11 paths are interleaved. And the red A->11 goes all the way to the bottom, but it could go to the top of the red bundle, resulting in less self-crossings. We should be able to find an algorithm that untangles these.

I won't hold up the release for this, as long as:
1. @jonmmease you can make an issue to revisit this later, and
2. @etpinard you won't consider it a breaking change if we implement this and make it the default sorting in the future.

Sound reasonable?

Reply:
> 2\. you won't consider it a breaking change if we implement this and make it the default sorting in the future.

It won't, I promise. :smirk:

Reply:
Yeah, I agree that we could certainly find a better default path sorting algorithm.  I'll make an issue to track this shortly.

Reply:
See https://github.com/plotly/plotly.js/issues/3059